### PR TITLE
594 carry forward funds update

### PR DESCRIPTION
**Pull request by xlorepdarkhelm**
## What changed

- Added new field to the CANFiscalYearCarryForward model, replacng amount with "received_amount" and "expected_amount".
- Changed "current_funding" to "received_funding" for CANFiscalYear model.
- Summing up the received_amount & expected_amount for carry forwards into a single value, also have the backend send the text for the carry-forward tag for the front-end, with the correct FY put in it.
- Refactored "carry_over" and "CarryOver" to "carry_forward" and "CarryForward" respectively.
- Added new Carry-Forward tag for can cards that appears if that can has any carry-forward amount in it for the current FY.

## Issue

[Carry-forward Funds Update #594](https://app.zenhub.com/workspaces/opre-ops-62d197b4c468970013bcd2ec/issues/gh/hhs/opre-ops/594)

## How to test

- Log into site
- Go to portfolios
- Choose "Healthy Marriage & Responsible Fatherhood"
- See that there is carry-forward in the portfolio summary bar.
- Scroll down through CANs, and 2 of them have carry-forward tags.

## Screenshots

362FC414-5BE1-4AAB-BDB8-929F9FE652CD

FB33830D-614D-43D5-AF05-2279C813EB61

### CODE REVIEW

File: backend/ops_api/ops/utils/cans.py [hunk 1]
```diff
@@ -1,43 +1,63 @@
 from typing import Optional, TypedDict
 
-from models.cans import CAN, BudgetLineItem, BudgetLineItemStatus, CANFiscalYear, CANFiscalYearCarryOver
+from models.cans import CAN, BudgetLineItem, BudgetLineItemStatus, CANFiscalYear, CANFiscalYearCarryForward
 
 
 class CanFundingSummary(TypedDict):
     """Dict type hint for total finding"""
```
Comment (tdonaworth) [inline backend/ops_api/ops/utils/cans.py:7]: comment typo - should be `funding` 😉

File: backend/ops_api/ops/utils/cans.py [hunk 2]
```diff
@@ -1,43 +1,63 @@
...
+    received_funding: float
     expected_funding: float
     total_funding: float
+    carry_forward_funding: float
+    carry_forward_label: str
     planned_funding: float
     obligated_funding: float
     in_execution_funding: float
-    available_funding: float
+    available_funding: str
```
Comment (tdonaworth) [inline backend/ops_api/ops/utils/cans.py:18]: Curious why this is a `str`.

File: frontend/src/components/CANs/CANFundingYTD/CANFundingYTD.jsx [hunk 3]
```diff
@@ -26,6 +33,14 @@ const CANFundingYTD = ({ total_funding = "0", current_funding = "0", expected_fu
     ];
     const [activeId, setActiveId] = React.useState(0);
 
+    const CarryForwardTag = ({ funding, label }) => {
+        if (funding > 0) {
```
Comment (tdonaworth) [inline frontend/src/components/CANs/CANFundingYTD/CANFundingYTD.jsx:37]: I forget, did we have a use-case where the carry forward could legit be negative (indicating an over spending)? I might be thinking of something else.

**Review by johndeange — APPROVED** [review]

File: backend/ops_api/ops/utils/cans.py (hunk 2 above)
Comment (xlorepdarkhelm) [reply backend/ops_api/ops/utils/cans.py:18]: I am not sure, we are doing formatting on the server side. I don't think that's what we should be doing. I can adjust it and make sure things work their way through correctly on the front-end.

File: backend/ops_api/ops/utils/cans.py (hunk 1 above)
Comment (xlorepdarkhelm) [reply backend/ops_api/ops/utils/cans.py:7]: Fixed

File: frontend/src/components/CANs/CANFundingYTD/CANFundingYTD.jsx (hunk 3 above)
Comment (xlorepdarkhelm) [reply frontend/src/components/CANs/CANFundingYTD/CANFundingYTD.jsx:37]: This is just rendering the tag for those cans that have carry-forward ( greater-than 0), vs those that don't (equal-to 0).

**Pull request merged**